exo-go/src/template_helpers/index.go

```diff
@@ -167,3 +168,49 @@ func CreateTmpServiceDir(chosenTemplate string) string {
 	}
 	return serviceTmpDir
 }
+
+// AddTemplate fetches a remote template from GitHub and stores it
+// under templateDir, returns an error if any
+func AddTemplate(gitURL, templateName, templateDir string) error {
+	if output, err := processHelpers.Run("", "git", "submodule", "add", "--name", templateName, gitURL, templateDir); err != nil {
+		return errors.Wrap(err, fmt.Sprintf("Failed to fetch template from %s: %s\n", gitURL, output))
+	}
+	return nil
+}
+
+// FetchTemplates fetches updates for all existing remote templates
+func FetchTemplates() error {
+	if output, err := processHelpers.Run("", "git", "submodule", "foreach", "git", "pull", "origin", "master"); err != nil {
+		return errors.Wrap(err, fmt.Sprintf("Failed to fetch updates for existing templates: %s\n", output))
+	}
+	return nil
+}
+
+// RemoveTemplate removes the given template from the application
+func RemoveTemplate(templateName, templateDir string) error {
+	if output, err := processHelpers.Run("", "git", "submodule", "deinit", "-f", templateDir); err != nil {
+		return errors.Wrap(err, fmt.Sprintf("Failed to deinit the template submodule: %s\n", output))
+	}
+	if output, err := processHelpers.Run("", "rm", "-rf", fmt.Sprintf(".git/modules/%s", templateName)); err != nil {
+		return errors.Wrap(err, fmt.Sprintf("Failed to force remove the template submodule: %s\n", output))
+	}
+	if output, err := processHelpers.Run("", "git", "rm", "-f", templateDir); err != nil {
+		return errors.Wrap(err, fmt.Sprintf("Failed to git remove the template submodule: %s\n", output))
+	}
+	return nil
+}
+
+// UpdateTemplate updates remote template according to the given GitHub URL,
+// return an error if any
+func UpdateTemplate(gitURL, templateName string) error {
+	if output, err := processHelpers.Run("", "git", "config", "--file=.gitmodules", fmt.Sprintf("submodule.%s.url", templateName), gitURL); err != nil {
+		return errors.Wrap(err, fmt.Sprintf("Failed to update git URL for template %s: %s\n", templateName, output))
+	}
+	if output, err := processHelpers.Run("", "git", "submodule", "sync"); err != nil {
+		return errors.Wrap(err, fmt.Sprintf("Failed to sync the submodule %s: %s\n", templateName, output))
+	}
+	if output, err := processHelpers.Run("", "git", "submodule", "update", "--init", "--recursive", "--remote"); err != nil {
+		return errors.Wrap(err, fmt.Sprintf("Failed to update the submodule %s: %s\n", templateName, output))
+	}
+	return nil
```

Comment:
Thoughts on creating a `runSeries` function which takes in an array of commands to run? On error it would wrap the error with
```go
fmt.Sprintf("Command Failed:\nCommand: %s\nOutput:\n%s\n\n'", command, output)
```